src/main/scala/Actor.scala

```diff
@@ -119,19 +119,19 @@ case class Actor(
     rookPath = rookPos <-> newRookPos
     mustBeUnoccupied = (kingPath ++ rookPath).filter(_ != kingPos).filter(_ != rookPos)
     if !mustBeUnoccupied.exists(board.pieces.contains)
-    if !kingPath.exists(p => board.variant.kingThreatened(board, !color, p))
-    b1 <- board take rookPos
-    b2 <- newKingPos match {
-      case p if p == kingPos => Some(b1)
-      case p => b1.move(kingPos, p)
-    }
-    b3 <- b2.place(color.rook, newRookPos)
-    b4 = b3 updateHistory (_ withoutCastles color)
+    b1 <- board take kingPos
```

Comment:
Do this first, to prevent the old king from blocking attacks to the squares that we will test. This did come up in standard chess, because if the old king positions blocks attacks, then we are in check anyway.